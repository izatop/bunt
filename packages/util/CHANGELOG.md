# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.11](https://github.com/izatop/typesafeunit/compare/v0.10.10...v0.10.11) (2020-11-19)


### Bug Fixes

* args ([5ed3806](https://github.com/izatop/typesafeunit/commit/5ed3806bcfef76265034072415cd09def14e3b48))





## [0.10.10](https://github.com/izatop/typesafeunit/compare/v0.10.9...v0.10.10) (2020-11-17)


### Bug Fixes

* formatters ([a2a01c6](https://github.com/izatop/typesafeunit/commit/a2a01c6698cb827ce0fecc03c31b5b704a9591c2))
* lint ([29a736b](https://github.com/izatop/typesafeunit/commit/29a736bf10587f41da49618a1bc7886ee9719f62))
* update ([d10e804](https://github.com/izatop/typesafeunit/commit/d10e804338153f07410ed2484b929f77a75c1b95))





## [0.10.7](https://github.com/izatop/typesafeunit/compare/v0.10.6...v0.10.7) (2020-11-11)


### Bug Fixes

* log ([f178fe4](https://github.com/izatop/typesafeunit/commit/f178fe4a3e18f5b6a8139f0a4987268f00344f60))
* logger ([315fba8](https://github.com/izatop/typesafeunit/commit/315fba84817522ba2a15b2af8fbc1b6e8704add8))
* logger ([8a0d30b](https://github.com/izatop/typesafeunit/commit/8a0d30b65512e225ad8c47940a7ade4a33ac278b))
* test ([adb105e](https://github.com/izatop/typesafeunit/commit/adb105e64a40456e663d635792af80fa3b32d6f9))





# [0.10.0](https://github.com/izatop/typesafeunit/compare/v0.9.12...v0.10.0) (2020-11-08)


### Features

* input package and new route system ([2914121](https://github.com/izatop/typesafeunit/commit/29141214a4dbfd0738f69dacf7aeda06578d12cf))





## [0.9.12](https://github.com/izatop/typesafeunit/compare/v0.9.11...v0.9.12) (2020-11-05)


### Bug Fixes

* validation log ([b99ad04](https://github.com/izatop/typesafeunit/commit/b99ad047d8afe15fb7ddc9a45143ee94c0a097ba))





## [0.9.11](https://github.com/izatop/typesafeunit/compare/v0.9.10...v0.9.11) (2020-11-05)


### Bug Fixes

* logger transport ([3ee4439](https://github.com/izatop/typesafeunit/commit/3ee44395193db16d65c0d59ea887960999c42e25))





## [0.9.10](https://github.com/izatop/typesafeunit/compare/v0.9.9...v0.9.10) (2020-11-04)


### Bug Fixes

* logger ([d983692](https://github.com/izatop/typesafeunit/commit/d98369266bfdff632f89da4db632a5c6784e46dd))





## [0.9.4](https://github.com/izatop/typesafeunit/compare/v0.9.3...v0.9.4) (2020-09-09)


### Bug Fixes

* lint ([f3095bc](https://github.com/izatop/typesafeunit/commit/f3095bc78a58fa765bbeff02114a9ca84619a682))





## [0.9.1](https://github.com/izatop/typesafeunit/compare/v0.9.0...v0.9.1) (2020-08-15)

**Note:** Version bump only for package @typesafeunit/util





# [0.9.0](https://github.com/izatop/typesafeunit/compare/v0.8.0...v0.9.0) (2020-08-15)


### Bug Fixes

* cors ([5ac872b](https://github.com/izatop/typesafeunit/commit/5ac872b96ffbe365e31339f1019b65943571e385))


### Features

* upgrade tests, deps and the web package ([6e8ba95](https://github.com/izatop/typesafeunit/commit/6e8ba958f0a215af2f9c8582b67d77e8a6df7c23))





# [0.8.0](https://github.com/izatop/typesafeunit/compare/v0.7.1...v0.8.0) (2020-08-03)

**Note:** Version bump only for package @typesafeunit/util





## [0.7.1](https://github.com/izatop/typesafeunit/compare/v0.7.0...v0.7.1) (2020-06-04)


### Bug Fixes

* stream ([9ab3d7e](https://github.com/izatop/typesafeunit/commit/9ab3d7e8cb8b9ccca11fd3e0ab3b9fa2be999abf))





# [0.7.0](https://github.com/izatop/typesafeunit/compare/v0.6.28...v0.7.0) (2020-05-26)


### Features

* new eslint rules ([62417eb](https://github.com/izatop/typesafeunit/commit/62417eb1385862c93108991075f864536b90a748))





## [0.6.28](https://github.com/izatop/typesafeunit/compare/v0.6.27...v0.6.28) (2020-05-17)


### Bug Fixes

* datetime ([95689bf](https://github.com/izatop/typesafeunit/commit/95689bf063707e996956c1d4ffcb5e96d98d0cf1))





## [0.6.27](https://github.com/izatop/typesafeunit/compare/v0.6.26...v0.6.27) (2020-05-15)


### Bug Fixes

* add locale, tz and calc begins/ends of a week ([ec31622](https://github.com/izatop/typesafeunit/commit/ec31622223b4680f0d7bebc36b99d65d366e0269))





## [0.6.26](https://github.com/izatop/typesafeunit/compare/v0.6.25...v0.6.26) (2020-05-12)


### Bug Fixes

* typescript 2.9 ([edea75d](https://github.com/izatop/typesafeunit/commit/edea75da40d97e96cbc74f3152f1bff79baa5225))





## [0.6.21](https://github.com/izatop/typesafeunit/compare/v0.6.20...v0.6.21) (2020-05-06)


### Bug Fixes

* datetime month mutation ([9ccb835](https://github.com/izatop/typesafeunit/commit/9ccb835fec7e77ee4e796893b96e8c35361f225b))





## [0.6.20](https://github.com/izatop/typesafeunit/compare/v0.6.19...v0.6.20) (2020-05-06)


### Bug Fixes

* datetime, update deps ([a2ccab2](https://github.com/izatop/typesafeunit/commit/a2ccab2c5dee781747d65ddceae7fb60524b0880))





## [0.6.19](https://github.com/izatop/typesafeunit/compare/v0.6.18...v0.6.19) (2020-05-01)


### Bug Fixes

* transform object -> transform input ([1fc2033](https://github.com/izatop/typesafeunit/commit/1fc20330c11cbb98583c75138ec2cff8f1a29692))





## [0.6.18](https://github.com/izatop/typesafeunit/compare/v0.6.17...v0.6.18) (2020-04-30)


### Bug Fixes

* logging ([2365a0b](https://github.com/izatop/typesafeunit/commit/2365a0b12902300b2e898e2c07aa5769be3b8e56))





## [0.6.12](https://github.com/izatop/typesafeunit/compare/v0.6.11...v0.6.12) (2020-04-01)


### Bug Fixes

* runtime und logger ([4f0cffe](https://github.com/izatop/typesafeunit/commit/4f0cffe0e63fc484ae99725a183283b7f3374839))





## [0.6.9](https://github.com/izatop/typesafeunit/compare/v0.6.8...v0.6.9) (2020-03-30)


### Bug Fixes

* web response ([0e5f9b6](https://github.com/izatop/typesafeunit/commit/0e5f9b6e45d0f8c8cfde5102aecdcb45ef2e417e))





## [0.6.7](https://github.com/izatop/typesafeunit/compare/v0.6.6...v0.6.7) (2020-03-30)


### Bug Fixes

* logger ([7ca423a](https://github.com/izatop/typesafeunit/commit/7ca423a085a38d524f1a5f0c2545ff95071367ad))





## [0.6.6](https://github.com/izatop/typesafeunit/compare/v0.6.5...v0.6.6) (2020-03-30)


### Bug Fixes

* logger owner name ([a8f2cf1](https://github.com/izatop/typesafeunit/commit/a8f2cf1ce886f6e5f376f652a7a1d5e1922391a6))





## [0.6.5](https://github.com/izatop/typesafeunit/compare/v0.6.4...v0.6.5) (2020-03-30)


### Bug Fixes

* log format ([409f834](https://github.com/izatop/typesafeunit/commit/409f834e428900d4088755fdbfcc2cc1614afaca))
* logger, runtime env ([c269a9e](https://github.com/izatop/typesafeunit/commit/c269a9e1f28f191b05655f4d0d26b9b4ea7624f6))





## [0.6.4](https://github.com/izatop/typesafeunit/compare/v0.6.3...v0.6.4) (2020-03-30)


### Bug Fixes

* log format ([8697069](https://github.com/izatop/typesafeunit/commit/86970698f70363b24478e3b1320c738d0b5fbbdc))





## [0.6.2](https://github.com/izatop/typesafeunit/compare/v0.6.1...v0.6.2) (2020-03-30)


### Bug Fixes

* logger ([cd5a36f](https://github.com/izatop/typesafeunit/commit/cd5a36fee0664015c99daab0f65f0f14d2409fcd))





## [0.6.1](https://github.com/izatop/typesafeunit/compare/v0.6.0...v0.6.1) (2020-03-30)


### Bug Fixes

* logger.perf ([ab7e561](https://github.com/izatop/typesafeunit/commit/ab7e561e3e9824a38749d1775abb7e39ec27024f))





# [0.6.0](https://github.com/izatop/typesafeunit/compare/v0.5.2...v0.6.0) (2020-03-30)


### Features

* logger, perf etc ([daed7fe](https://github.com/izatop/typesafeunit/commit/daed7fe86ea6e9cb3bd90830a77167e6f9e21755))





# [0.5.0](https://github.com/izatop/typesafeunit/compare/v0.4.6...v0.5.0) (2020-03-26)


### Features

* improve flow control ([1d4a31d](https://github.com/izatop/typesafeunit/commit/1d4a31da7b1801fa7eb66e5a7b699a1b79cede2b))





## [0.4.2](https://github.com/izatop/typesafeunit/compare/v0.4.1...v0.4.2) (2020-03-25)

**Note:** Version bump only for package @typesafeunit/util





## [0.4.1](https://github.com/izatop/typesafeunit/compare/v0.4.0...v0.4.1) (2020-03-25)

**Note:** Version bump only for package @typesafeunit/util





# [0.4.0](https://github.com/izatop/typesafeunit/compare/v0.3.9...v0.4.0) (2020-03-25)


### Features

* http server ([62abc0f](https://github.com/izatop/typesafeunit/commit/62abc0f20cfefac0a7d5fa6bc91c030719d5e9a8))





## [0.3.9](https://github.com/izatop/typesafeunit/compare/v0.3.8...v0.3.9) (2020-03-08)


### Bug Fixes

* **validation:** skip validation if not required ([d50b893](https://github.com/izatop/typesafeunit/commit/d50b8936a26445b3e098fad01a8c6b0d23933081))





## [0.3.6](https://github.com/izatop/typesafeunit/compare/v0.3.5...v0.3.6) (2020-03-08)

**Note:** Version bump only for package @typesafeunit/util





## [0.3.5](https://github.com/izatop/typesafeunit/compare/v0.3.4...v0.3.5) (2020-03-07)


### Bug Fixes

* **headers:** case sensitivity fix ([c119eae](https://github.com/izatop/typesafeunit/commit/c119eae6fb96fba11f27c260f130b351addc34fa))





## [0.3.4](https://github.com/izatop/typesafeunit/compare/v0.3.3...v0.3.4) (2020-03-07)

**Note:** Version bump only for package @typesafeunit/util





## [0.3.3](https://github.com/izatop/typesafeunit/compare/v0.3.2...v0.3.3) (2020-03-05)


### Bug Fixes

* **npm:** add LICENSE ([d7bf57f](https://github.com/izatop/typesafeunit/commit/d7bf57fe2dcb9f83cc4a570c99fafcb5e318bdd4))





## [0.3.2](https://github.com/izatop/typesafeunit/compare/v0.3.1...v0.3.2) (2020-03-05)

**Note:** Version bump only for package @typesafeunit/util





## [0.3.1](https://github.com/izatop/typesafeunit/compare/v0.3.0...v0.3.1) (2020-03-05)

**Note:** Version bump only for package @typesafeunit/util





# 0.3.0 (2020-03-05)


### Features

* initial commit ([a76a484](https://github.com/izatop/typesafeunit/commit/a76a4849282c382311f4d0cb9f9bd66a10bcceaa))
