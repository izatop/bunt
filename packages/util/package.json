{
    "name": "@bunt/util",
    "version": "0.24.29",
    "keywords": [
        "typescript"
    ],
    "author": {
        "name": "Artur Bier",
        "email": "izatop@gmail.com"
    },
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "files": [
        "dist/"
    ],
    "description": "Utils",
    "repository": "git@github.com:izatop/bunt.git",
    "scripts": {
        "clean": "yarn build:clean",
        "watch": "yarn build:watch",
        "build": "yarn build:clean && tsc",
        "build:clean": "rimraf dist tsconfig.tsbuildinfo",
        "build:watch": "yarn build --watch"
    },
    "publishConfig": {
        "access": "public"
    },
    "license": "MIT"
}
