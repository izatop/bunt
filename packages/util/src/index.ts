export * from "./is";
export * from "./stream";
export * from "./assert";
export * from "./object";
export * from "./string";
export * from "./decorator";
export * from "./function";
export * from "./callback";
export * from "./Date";
export * from "./Transform";
export * from "./Logger";
export * from "./Exception";
export * from "./Node";
export * from "./Perf";
export * from "./Async";
export * from "./Ref";
export * from "./Map";
export * from "./interfaces";
