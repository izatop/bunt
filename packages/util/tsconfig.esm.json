{
  "extends": "../../tsconfig.lib.json",
  "compilerOptions": {
    "module": "ESNext",
    "moduleResolution": "Node",
    "rootDir": "src",
    "outDir": "dist/esm"
  },
  "references": [
    {
      "path": "../assert/tsconfig.esm.json"
    },
    {
      "path": "../type/tsconfig.esm.json"
    },
    {
      "path": "../async/tsconfig.esm.json"
    },
    {
      "path": "../is/tsconfig.esm.json"
    }
  ],
  "include": [
    "src"
  ]
}
