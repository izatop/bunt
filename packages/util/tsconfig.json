{
  "extends": "../../tsconfig.lib.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "dist/cjs"
  },
  "references": [
    {
      "path": "../assert"
    },
    {
      "path": "../type"
    },
    {
      "path": "../async"
    },
    {
      "path": "../is"
    }
  ],
  "include": [
    "src"
  ]
}
