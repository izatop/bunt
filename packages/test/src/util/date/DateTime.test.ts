import {DateTime, DateTimeKind} from "@typesafeunit/util";

describe("DateTime", () => {
    const date = new DateTime("2020-02-12T12:35:13.123Z");
    test("mutate()", () => {
        const mutations: [DateTimeKind, number][] = [
            ["ms", +100],
            ["sec", +10],
            ["min", -10],
            ["hour", +2],
            ["day", -1],
            ["year", -5],
        ];

        const result = [];
        for (const value of mutations) {
            result.push(date.mutate(value).getDate());
        }

        expect(result).toMatchSnapshot();
    });

    test("set()", () => {
        const setters: [DateTimeKind, number][] = [
            ["ms", 333],
            ["sec", 22],
            ["min", 11],
            ["hour", 0],
            ["day", 11],
            ["year", 2022],
        ];

        const result = [];
        for (const value of setters) {
            result.push(date.set(value).getDate());
        }

        expect(result).toMatchSnapshot();
    });

    test("begins/ends()", () => {
        const result = [];
        const begins: Exclude<DateTimeKind, "ms" | "week">[] = ["sec", "min", "hour", "day", "month", "year"];
        for (const kind of begins) {
            result.push({
                begins: date.begins(kind).getDate(),
                ends: date.ends(kind).getDate(),
            });
        }

        expect(result).toMatchSnapshot();
    });
});
