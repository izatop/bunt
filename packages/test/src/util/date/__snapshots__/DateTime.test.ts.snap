// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime begins/ends() 1`] = `
Array [
  Object {
    "begins": 2020-02-12T12:35:13.000Z,
    "ends": 2020-02-12T12:35:13.999Z,
  },
  Object {
    "begins": 2020-02-12T12:35:00.000Z,
    "ends": 2020-02-12T12:35:59.999Z,
  },
  Object {
    "begins": 2020-02-12T12:00:00.000Z,
    "ends": 2020-02-12T12:59:59.999Z,
  },
  Object {
    "begins": 2020-02-11T21:00:00.000Z,
    "ends": 2020-02-12T20:59:59.999Z,
  },
  Object {
    "begins": 2020-01-30T21:00:00.000Z,
    "ends": 2020-02-29T20:59:59.999Z,
  },
  Object {
    "begins": 2019-12-30T21:00:00.000Z,
    "ends": 2020-12-31T20:59:59.999Z,
  },
]
`;

exports[`DateTime mutate() 1`] = `
Array [
  2020-02-12T12:35:13.223Z,
  2020-02-12T12:35:23.123Z,
  2020-02-12T12:25:13.123Z,
  2020-02-12T14:35:13.123Z,
  2020-02-11T12:35:13.123Z,
  2015-02-12T12:35:13.123Z,
]
`;

exports[`DateTime set() 1`] = `
Array [
  2020-02-12T12:35:13.333Z,
  2020-02-12T12:35:22.123Z,
  2020-02-12T12:11:13.123Z,
  2020-02-11T21:35:13.123Z,
  2020-02-11T12:35:13.123Z,
  2022-02-12T12:35:13.123Z,
]
`;
